src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -146,13 +138,52 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 }
             });
 
+        // These events might all be triggered by the connection, but that
+        // should be drained and closed before we complete destruction. So these
+        // are safe.
+        //
+        // NOTE: _ScrollPositionChanged has to be registered after we set up the
+        // _updateScrollBar func. Otherwise, we could get a callback from an
+        // attached content before we set up the throttled func, and that'll A/V
+        _revokers.coreScrollPositionChanged = _core.ScrollPositionChanged(winrt::auto_revoke, { get_weak(), &TermControl::_ScrollPositionChanged });
+        _revokers.WarningBell = _core.WarningBell(winrt::auto_revoke, { get_weak(), &TermControl::_coreWarningBell });
+        _revokers.CursorPositionChanged = _core.CursorPositionChanged(winrt::auto_revoke, { get_weak(), &TermControl::_CursorPositionChanged });
+
         static constexpr auto AutoScrollUpdateInterval = std::chrono::microseconds(static_cast<int>(1.0 / 30.0 * 1000000));
         _autoScrollTimer.Interval(AutoScrollUpdateInterval);
         _autoScrollTimer.Tick({ get_weak(), &TermControl::_UpdateAutoScroll });
 
         _ApplyUISettings();
     }
 
+    Control::TermControl TermControl::AttachContent(Control::ControlInteractivity content, const Microsoft::Terminal::Control::IKeyBindings& keyBindings)
+    {
+        const auto term{ winrt::make_self<TermControl>(content) };
+        term->_AttachDxgiSwapChainToXaml(reinterpret_cast<HANDLE>(term->_core.SwapChainHandle()));
+        content.Reparent(keyBindings);

```

Comment:
This is one of those things that's not perfectly refactored. It's "good enough". Literally all of the settings that are passed to the ControlCore are copied out of the settings, except for the keybindings. For all other settings, ControlCore owns its own little copy of the settings relevant to itself. 

The `Page` ultimately owns the Keybindings. When we build a TermControl, we pass in the page's KB's as the `IKeybindings` in the terminal settings. The Core ends up being the thing that owns the settings (and therefore  has a reference to the keybindings of the page that created it). This is wacky and kinda always has been. 

So now we've got an issue though - if you reparent a ControlCore to a new page, it can't be calling keybindings from it's previous page! That'd cause it to like, open new tabs in the original window :fear:

Does this make any sense?